src/dash/models/DashManifestModel.js

```diff
@@ -292,6 +292,24 @@ function DashManifestModel(config) {
         return adaptation[DashConstants.CENC_DEFAULT_KID];
     }
 
+    function getLabelForAdaptation(adaptation) {
+        if (adaptation.lang && !adaptation.Label_asArray) {
+            return adaptation.lang;
+        }
+
+        if (adaptation.lang && adaptation.Label_asArray) {
```

Comment:
Question, if there's no `Label`s, will `adaptation.Label_asArray` be `[]` or `undefined`? I ask because `[]` is truthy, so you could do `if (adaptation.lang && adaptation.Label_asArray.length) {` to only enter the code block when the array actually has elements in it.

If you only enter the code block when there's at least 1 element, I don't think you need the `return adaptation.lang` in it.

Reply:
@squarebracket It will be `undefined`